### ISSUE TITLE
[MU4 Task] New command for "Apply current input state"

### ISSUE THREAD

**Issue by bkunda** [musescore/MuseScore#11517]
**Task description**
See https://github.com/musescore/MuseScore/pull/5505 for background information on this one.
Also relates to discussion in https://github.com/musescore/MuseScore/issues/11500

1. In input mode only, a new `Apply current input state` command would look at the selected note duration, as well as any accidentals, and apply them to the note that has just been entered.

Example: 

- User is in note input mode
- The note duration selected is a quarter note
- The **sharp** accidental button is also selected

<img width="1041" alt="Screenshot 2022-05-05 at 1 07 43 am" src="https://user-images.githubusercontent.com/86290556/166711938-0f089aa9-0b75-4410-92f6-23f2427945fe.png">

In this example, executing the `Apply current input state` command would apply these values would apply to the _previous note_; I.e. The first note on the pictured staff would become a C-sharp.

Importantly, the note input cursor would remain unaffected, and the user would remain in note input mode. 

Note: The behaviour for this already exists in the commands `Note input (non-toggle): sharp` (and likewise for "flat" and "natural" variants). These commands have not been ported from MS3.6 to MS4 and would not need to be if the present task is able to be implemented.

2. We suggest assigning by default the shortcut `Y` to this command.

3. As a related issue, we would like the shortcut `Shift` + `Y` to execute the existing command `Apply current palette element`.

Please let me or @MarcSabatella know if there are any questions or concerns about this one. 

And thanks very much!

**Comment by Tantacrul** [comment]
If this is a feature request (rather than a regression fix), I suggest we move it to 4.x

**Comment by MarcSabatella** [comment]
It’s a regression in that for MuseScore 3, Q and W served to apply retroactively but don’t anymore.  I am also proposing we consider removing the confusing second retroactive set of accidental commands and replace them with this.  And that feels like a change better suited than 4.0 than 4.x.  But anyhow, it will be extremely welcome whenever it comes.

**Comment by Tantacrul** [comment]
Was it lost in 3.6? Because I can't see to do it in that build.

**Comment by MarcSabatella** [comment]
In all MuseScore 3 versions, pressing "Q" or "W" immediately after entering a note halves or doubles the duration, thus providing a fix if you forget to change duration before entering a note.  This was changed a few months ago for MuseScore 4 to instead work more like the duration commands in setting duration "preemptively" instead of "retroactively".  It kind of makes sense for consistency, but I think the real reason was to provide a second method of setting duration for tablature, instead of bleeding Shift+number.  I was never convinced that was an improvement, but I can see the value of consistency here and let it go.

What we lost, though, was an easy way of changing duration retroactively.  We faced exactly the same issue around 3.3 or so when we changed the behavior of accidental commands to work preemptively instead of retroactively.  The backlash was fast and furious, so we eventually added back the retroactive ("non-toggle") versions of those commands.  But right now those retroactive commands are gone also.  I'm not sure if because they were never ported or because they have no default shortcut and thus are prevented from appearing the in the shortcut dialog by the bug that prevents *all* commands without shortcuts from appearing.

But bottom line, 3.6.2 had ways of changing duration or adding an accidental after entering a note, and MU4 currently lacks these.  I don't think the *way* MU3 supported these was particularly elegant, and both were certainly completely non-discoverable and required memorizing five different shortcuts (Q, W, and custom shortcuts you needed to define for retroactive-flat, retroactive-sharp, retroactive-natural).  Adding this command restores the functionality lost from 3.6.2, and does it in a far more elegant way that needs only a single shortcut (which @bkunda and I were currently discussing be "Y" for "applY" but are completely open to alternatives).

To be fair, there *is* still a way to change duration retroactively in MU4, but it's twice as much work as MU3: use Shift+Q/W, which goes through the dotted durations too.  But to change a whole note to an eighth note took three keystrokes in MU3 (Q Q Q); it would take six in MU4 (Shift+Q Shift+Q Shift+Q Shift+Q Shift+Q Shift+Q); it takes two with this new feature (4 Y).  Similarly, if "-" becomes the command to toggle the flat button (which is the current proposal), then doing so retroactively just requires hitting "Y" afterward.

I would also propose a toolbar button be made available for this command, perhaps not installed by default, but that much could wait longer.
